### ISSUE TITLE
cant seem to get the wildcard feature to work while blacklisting nodes

### ISSUE THREAD

**Issue by CalvinWong-Reflektion** [AcalephStorage/consul-alerts#174]
i am able to blacklist single nodes, but not wild card.

single node -
```
NFO[6640] consul-client005:_:check-api is blacklisted.
INFO[6640] consul-client005:_:check-disk is blacklisted.
INFO[6640] consul-client005:_:check-file is blacklisted.
INFO[6640] consul-client005:_:check-interface is blacklisted.
INFO[6640] consul-client005:_:check-proc is blacklisted.
INFO[6640] consul-client005:_:check-tcp is blacklisted.
INFO[6640] consul-client005:_:serfHealth is blacklisted.
```

this is my K/V for wildcard. not sure if the pattern is correct -
```
consul-client.*
true
```

**Comment by rokka-n** [comment]
Same here, blacklist for nodes is ignored even if exact name is given

![image](https://cloud.githubusercontent.com/assets/4412410/25410795/bcaa8128-29cd-11e7-9970-3baffa0faf8d.png)

**Comment by nidhishopback** [comment]
Hi,

Has anyone been able to get wildcard working to blacklist nodes?

**Comment by Gerrrr** [comment]
Hi,

Yes, it works perfectly for us, e.g. `["^.*-ignore"$]`.

**Comment by CalvinWong-Reflektion** [comment]
it didnt seem to work for me.
i have the following nodes -
```
NAME               ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
consul-client001   -        virtualbox   Running   tcp://192.168.99.101:2376           v17.04.0-ce
consul-client002   -        virtualbox   Running   tcp://192.168.99.102:2376           v17.04.0-ce
consul-client003   -        virtualbox   Running   tcp://192.168.99.103:2376           v17.04.0-ce
consul-client004   -        virtualbox   Running   tcp://192.168.99.104:2376           v17.04.0-ce
consul-client005   -        virtualbox   Running   tcp://192.168.99.105:2376           v17.04.0-ce
consul-server001   -        virtualbox   Running   tcp://192.168.99.100:2376           v17.04.0-ce
```

i have this -
```
["^.*-client00.*"$]
```
but doesnt seem to blacklist anything.

i kept seeing this line -
```
INFO[2352] Regexp matching host found (^.*-client.*$).
```

**Comment by rokka-n** [comment]
What happens if pattern is .+ instead of .*?

**Comment by CalvinWong-Reflektion** [comment]
looks to be the same. no luck.

so it works for you guys? i must be doing something wrong!! :-\

its under -
```
consul-alerts/config/checks/blacklist/nodes/
```
right? LOL

**Comment by Gerrrr** [comment]
@CalvinWong-Reflektion it must be valid JSON, sorry for the mistake I made in the previous response. Can you please try `["^.*-client00.*$"]`?

**Comment by CalvinWong-Reflektion** [comment]
no luck from my end.

what does the log output show when you enable regex blacklist?

for single host i actually see the log saying its being blacklisted -
```
INFO[76570] consul-client003:_:check-api is blacklisted.
INFO[76570] consul-client004::TCP on port 8080 is now pending status change from passing to critical.
INFO[76570] consul::TCP on port 8080 is now back to passing.
INFO[76570] consul-client003:_:check-disk is blacklisted.
INFO[76570] consul-client003:_:check-file is blacklisted.
INFO[76570] consul-client003:_:check-interface is blacklisted.
INFO[76570] consul-client003:_:check-proc is blacklisted.
INFO[76570] consul-client003:_:check-tcp is blacklisted.
INFO[76570] consul-client003:_:serfHealth is blacklisted.
```

**Comment by Gerrrr** [comment]
Hey @CalvinWong-Reflektion 
Just figured that we are running a bit outdated fork of consul-alerts in production, so there might be some difference from your setup. I am going to try the blacklisting feature against the master branch during the weekend.

**Comment by shopback-travis** [comment]
Does not work for me too. I am also using the latest code from consul-alerts

**Comment by nidhishopback** [comment]
Does not work for me too
I am using the latest code from consul-alerts

**Comment by Gerrrr** [comment]
Hey guys,

Sorry that it took me longer than expected. I have just tried the following setup with a bit outdated Consul and the latest consul-alerts built from the master.
**NB** the node name and the regular expression on the screenshot.

```
$ consul agent -dev
==> Starting Consul agent...
==> Starting Consul agent RPC...
==> Consul agent running!
           Version: 'v0.7.2'
         Node name: 'Sorokoumovs-MBP'
        Datacenter: 'dc1'
            Server: true (bootstrap: false)
       Client Addr: 127.0.0.1 (HTTP: 8500, HTTPS: -1, DNS: 8600, RPC: 8400)
      Cluster Addr: 127.0.0.1 (LAN: 8301, WAN: 8302)
    Gossip encrypt: false, RPC-TLS: false, TLS-Incoming: false
             Atlas: <disabled>

==> Log data will now stream in as it occurs:

    2017/05/14 16:55:55 [INFO] raft: Initial configuration (index=1): [{Suffrage:Voter ID:127.0.0.1:8300 Address:127.0.0.1:8300}]
    2017/05/14 16:55:55 [INFO] raft: Node at 127.0.0.1:8300 [Follower] entering Follower state (Leader: "")
    2017/05/14 16:55:55 [INFO] serf: EventMemberJoin: Sorokoumovs-MBP 127.0.0.1
    2017/05/14 16:55:55 [INFO] consul: Adding LAN server Sorokoumovs-MBP (Addr: tcp/127.0.0.1:8300) (DC: dc1)
    2017/05/14 16:55:55 [INFO] serf: EventMemberJoin: Sorokoumovs-MBP.dc1 127.0.0.1
    2017/05/14 16:55:55 [INFO] consul: Adding WAN server Sorokoumovs-MBP.dc1 (Addr: tcp/127.0.0.1:8300) (DC: dc1)
```

<img width="765" alt="screen shot 2017-05-14 at 16 59 28" src="https://cloud.githubusercontent.com/assets/918393/26035094/2574c8c8-38c7-11e7-9aac-0193bfe19d49.png">

```
$ build/bin/Darwin-i386/latest/consul-alerts start --log-level=debug --watch-checks --watch-events
INFO[0000] Checking consul agent connection...
INFO[0000] Sorokoumovs-MBP:_:serfHealth is blacklisted.
INFO[0000] Consul Alerts daemon started
INFO[0000] Consul Alerts Host: Sorokoumovs-MBP
INFO[0000] Consul Agent: localhost:8500
INFO[0000] Consul Datacenter: dc1
INFO[0000] Started Consul-Alerts API
INFO[0000] Running for leader election...
INFO[0000] Now acting as leader.
INFO[0000] Now watching for health changes.
INFO[0000] Now watching for events.
```

Then I started Nomad in dev mode to get some new healthchecks to be blacklisted:

```
$ nomad agent -dev
    No configuration files loaded
==> Starting Nomad agent...
==> Nomad agent configuration:

                 Atlas: <disabled>
                Client: true
             Log Level: DEBUG
                Region: global (DC: dc1)
                Server: true
               Version: 0.5.4

==> Nomad agent started! Log data will stream in below:
```

... and observed the following output from consul-alerts:

```
INFO[0011] Sorokoumovs-MBP:_nomad-server-nomad-serf:09df7ac0bea48c4737c78a55c3d62f1560a3ef33 is blacklisted.
INFO[0011] Sorokoumovs-MBP:_nomad-server-nomad-http:275647020c41fc9cc49257cf60b7822f5cb2d46d is blacklisted.
INFO[0011] Sorokoumovs-MBP:_nomad-server-nomad-rpc:3ba98d57c52e60b07bb14d3250b5a36977c6ea0f is blacklisted.
INFO[0011] Sorokoumovs-MBP:_nomad-client-nomad-client-http:a8787e906608c95608245cf9a1522cc54a60734f is blacklisted.
INFO[0011] Sorokoumovs-MBP:_:serfHealth is blacklisted.
INFO[0021] Sorokoumovs-MBP:_nomad-server-nomad-serf:09df7ac0bea48c4737c78a55c3d62f1560a3ef33 is blacklisted.
INFO[0021] Sorokoumovs-MBP:_nomad-server-nomad-http:275647020c41fc9cc49257cf60b7822f5cb2d46d is blacklisted.
INFO[0021] Sorokoumovs-MBP:_nomad-server-nomad-rpc:3ba98d57c52e60b07bb14d3250b5a36977c6ea0f is blacklisted.
INFO[0021] Sorokoumovs-MBP:_nomad-client-nomad-client-http:a8787e906608c95608245cf9a1522cc54a60734f is blacklisted.
INFO[0021] Sorokoumovs-MBP:_:serfHealth is blacklisted.
```

Can you please confirm that you put a list of regular expressions under the key `consul-alerts/config/checks/blacklist/nodes` (as in the example above) or specify the individual nodes - `consul-alerts/config/checks/blacklist/nodes/<exact-node-name-to-be-blacklisted>` (e.g. `consul-alerts/config/checks/blacklist/nodes/node1`, `consul-alerts/config/checks/blacklist/nodes/node2`)?

**Comment by CalvinWong-Reflektion** [comment]
![image](https://cloud.githubusercontent.com/assets/22897932/26071348/99c0a860-395c-11e7-9eb3-14566441ce0d.png)
oh boy. i was such an idiot!!!
`consul-alerts/config/checks/blacklist/nodes` is a key.. not a folder!!!
i had the following -
- folder: nodes
- key: `["^.*-client00.*$"]`
- value: `true`

now i see that its being blacklisted!!! \(ᵔ ³ᵔ)/
```
INFO[1500] consul-client005:_:check-api is blacklisted.
INFO[1500] consul-client005:_:check-tcp is blacklisted.
INFO[1500] consul-client005:_:check-disk is blacklisted.
INFO[1500] consul-client005:_:check-file is blacklisted.
INFO[1500] consul-client005:_:check-interface is blacklisted.
INFO[1500] consul-client005:_:check-proc is blacklisted.
INFO[1500] consul-client005:_:serfHealth is blacklisted.
```

**Comment by nidhishopback** [comment]
Hi
I can't seem to make it work still..Am I missing something here?

![screen shot 2017-05-16 at 10 16 19 am](https://cloud.githubusercontent.com/assets/21728733/26087366/ea8923f0-3a22-11e7-965f-1deb1a700798.png)

consul version - 0.7.2

Nidhi

**Comment by CalvinWong-Reflektion** [comment]
@nidhishopback i wonder if the `$` is the regular expression might make any diff
`["^.*worker.*$"]`

**Comment by Gerrrr** [comment]
@CalvinWong-Reflektion don't worry! From the number of people, who face issues with disabling health checks, it seems that the corresponding README section can be better. I'll think how to improve it.

@nidhishopback what version of consul-alerts do you run? Unfortunately, the wildcard feature did not appear yet in the latest release and is only available in master.

**Comment by rokka-n** [comment]
@Gerrrr few screenshots would help 😛
I had troubles myself figuring it out. I mean managing app config from consul KV is a whole new idea to most developers.

**Comment by nidhishopback** [comment]
@Gerrrr I am using Consul Alerts 0.3.3. I have set up using chef cookbook, which picks up the binary from here->

'https://bintray.com/artifact/download/darkcrux/generic/'

Are you saying that the consul-alerts wildcard feature is not present in v0.4.0 as well? Do you know when its going to be released?

**Comment by Gerrrr** [comment]
@nidhishopback unfortunately, this feature is not available in v0.4.0. Can't tell you the exact date, but I am going to spend some time over the weekend to review some PRs and prepare for the next release.

**Comment by nidhishopback** [comment]
@Gerrrr Thanks for the update.
It would be great if you could shared the compiled consul-alerts binary of the latest master, where wildcards worked.
I tried compiling it myself, but it wouldn't start

**Comment by Gerrrr** [comment]
Hey @nidhishopback we released v0.5.0!  You can find binaries at the [release page](https://github.com/AcalephStorage/consul-alerts/releases/tag/v0.5.0).

**Comment by nidhishopback** [comment]
Thank you so much!! The wildcards are working now :)
